uplink/src/collector/downloader.rs

```diff
@@ -312,42 +324,17 @@ impl FileDownloader {
     async fn download(
         &mut self,
         resp: Response,
-        mut file: File,
-        content_length: usize,
+        download: &mut DownloadState,
```

Comment:
Make `DownloadState` a member of `FileDownloader`. This is a mutable method with side-effects any way. No need to pass an separate object as argument for a different state

Add clear method to `DownloadState` to reset on finishing download